### PR TITLE
DEV-3566 fix virus interpretation

### PR DESCRIPTION
**Pull request by mvanniekerkHartwig**
Removes the `VirusInterpretation` enum. During the orange-datamodel creation, I added this enum because it was part of the oncoact (patient-reporter/rose/protect) common module. In hindsight not a smart move, the `VirusInterpretation` enum was removed from hmftools originally all the way back in 2021 (see commit `3de88bcd`). There seems to be no reason that this enum should exist, Paul already fixed a bug related to this same field in May (see commit `45319b5b`).

### CODE REVIEW

**Review by kduyvesteyn — COMMENTED** [review]
Are you sure you added this in orange-datamodel creation?

In any case, we use and rely on this field in ACTIN since nov 2022, and we explicitly use the various enum values. If we change this to a string we would map it back to enum in ACTIN, not sure if that's an improvement?

**Review by DFKoetsier — APPROVED** [review]
Looks great!

**Comment by mvanniekerkHartwig** [comment]
> Are you sure you added this in orange-datamodel creation?

Yes, in commit `d58e42d9`.

> In any case, we use and rely on this field in ACTIN since nov 2022, and we explicitly use the various enum values. If we change this to a string we would map it back to enum in ACTIN, not sure if that's an improvement?

Agreed, that would not be ideal. On the other hand, nowhere in hmftools it is guaranteed that the values in `VirusInterpretation` will be the only values in the `AnnotatedVirus.interpretation` field. As long as this is not guaranteed on the side of the virus output, I'd argue relying on this enum is a bug. Another practical consideration is that it's currently much easier to update ACTIN than Orange.
Patient Reporter has the same problem but mostly checks for `virus.interpretation() == VirusInterpretation.HPV`, and when changing this to the hardcoded String value "HPV" we probably don't loose out in code reliability too much.

**Comment by kduyvesteyn** [comment]
> > Are you sure you added this in orange-datamodel creation?
> 
> Yes, in commit `d58e42d9`.
> 
> > In any case, we use and rely on this field in ACTIN since nov 2022, and we explicitly use the various enum values. If we change this to a string we would map it back to enum in ACTIN, not sure if that's an improvement?
> 
> Agreed, that would not be ideal. On the other hand, nowhere in hmftools it is guaranteed that the values in `VirusInterpretation` will be the only values in the `AnnotatedVirus.interpretation` field. As long as this is not guaranteed on the side of the virus output, I'd argue relying on this enum is a bug. Another practical consideration is that it's currently much easier to update ACTIN than Orange. Patient Reporter has the same problem but mostly checks for `virus.interpretation() == VirusInterpretation.HPV`, and when changing this to the hardcoded String value "HPV" we probably don't loose out in code reliability too much.

Ah I see now, didn't realize interpretation() is a string in VirusInterpreter. Maybe the best solution would be to make this an enum in VirusInterpreter and propagate that. Downstream tools should be able to know the list of potential viruses assigned after interpretation, and not have to guess it. 

In similar situations we added extra enums all the way rather than remove "wrong" ones. E.G. regionType and codingType in LinxBreakend were both changed from String to enum after realising downstream tools need to be able to know what types could be assigned.

**Comment by kduyvesteyn** [comment]
Do note: If you decide to make interpretation a string all-the-way then feel free to merge (code changes themselves look good, and do indeed solve an inconsistency in ORANGE)

**Comment by mvanniekerkHartwig** [comment]
> In similar situations we added extra enums all the way rather than remove "wrong" ones. E.G. regionType and codingType in LinxBreakend were both changed from String to enum after realising downstream tools need to be able to know what types could be assigned.

I did a small investigation to see what it would take to go all the way. Ultimately, the `AnnotatedVirus.interpretation` field is set by a CSV file in virus interpreter that is passed in as an input file through the command line arg `virus_reporting_db_tsv`. It seems doable, so I made the changes. No complete guarantee of correctness, because now the change is a bit more sophisticated. But pretty high confidence it should work.

File: hmf-common/src/main/java/com/hartwig/hmftools/common/virus/VirusConstants.java
Comment (kduyvesteyn) [inline hmf-common/src/main/java/com/hartwig/hmftools/common/virus/VirusConstants.java:1]: I'd rename to `VirusType` or `VirusInterpretation`

Current name doesn't sound intuitive, I expected I'd find a few public static Strings but not an enum!

**Review by kduyvesteyn — APPROVED** [review]
Looks good! Some smallish comments

File: hmf-common/src/main/java/com/hartwig/hmftools/common/virus/AnnotatedVirusFile.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ private static String toString(@NotNull AnnotatedVirus annotatedVirus)
                 .add(annotatedVirus.name())
                 .add(annotatedVirus.qcStatus().toString())
                 .add(String.valueOf(annotatedVirus.integrations()))
-                .add(annotatedVirus.interpretation())
+                .add(String.valueOf(annotatedVirus.interpretation()))
```
[the remark this answers is not in the document]
Comment (mvanniekerkHartwig) [reply hmf-common/src/main/java/com/hartwig/hmftools/common/virus/AnnotatedVirusFile.java:126]: For this one, I tried to stick as closely as possible to current behaviour. Currently, we write the String value "null" if interpretation is `null`. I'd like to keep current behaviour so we only need to update orange and not virus-interpreter as well.

**Review by DFKoetsier — CHANGES_REQUESTED** [review]
I found a minor issue

File: patient-db/src/main/java/com/hartwig/hmftools/patientdb/dao/VirusInterpreterDAO.java [hunk 2]
```diff
@@ -54,7 +54,7 @@ private static void addVirusAnnotation(@NotNull Timestamp timestamp, @NotNull In
                 annotatedVirus.name(),
                 annotatedVirus.qcStatus(),
                 annotatedVirus.integrations(),
-                annotatedVirus.interpretation(),
+                String.valueOf(annotatedVirus.interpretation()),
```
[the remark this answers is not in the document]
Comment (mvanniekerkHartwig) [reply patient-db/src/main/java/com/hartwig/hmftools/patientdb/dao/VirusInterpreterDAO.java:57]: Good catch!

File: cuppa/src/main/java/com/hartwig/hmftools/cup/feature/FeatureDataLoader.java [hunk 3]
```diff
@@ -306,7 +307,7 @@ private static Map<String,List<AnnotatedVirus>> getAllViruses(final DatabaseAcce
                         .taxid(record.getValue(VIRUSANNOTATION.TAXID))
                         .name(record.getValue(VIRUSANNOTATION.VIRUSNAME))
                         .qcStatus(VirusBreakendQCStatus.valueOf(record.getValue(VIRUSANNOTATION.QCSTATUS)))
-                        .interpretation(interpretation)
+                        .interpretation(VirusConstants.fromVirusName(interpretation))
```
[the remark this answers is not in the document]
Comment (mvanniekerkHartwig) [reply cuppa/src/main/java/com/hartwig/hmftools/cup/feature/FeatureDataLoader.java:310]: good catch!

File: cuppa/src/main/java/com/hartwig/hmftools/cup/feature/FeatureDataLoader.java [hunk 4]
```diff
@@ -306,7 +307,7 @@ private static Map<String,List<AnnotatedVirus>> getAllViruses(final DatabaseAcce
                         .taxid(record.getValue(VIRUSANNOTATION.TAXID))
                         .name(record.getValue(VIRUSANNOTATION.VIRUSNAME))
                         .qcStatus(VirusBreakendQCStatus.valueOf(record.getValue(VIRUSANNOTATION.QCSTATUS)))
-                        .interpretation(interpretation)
+                        .interpretation(interpretation == null ? null : VirusType.fromVirusName(interpretation))
```
[the remark this answers is not in the document]
Comment (mvanniekerkHartwig) [reply cuppa/src/main/java/com/hartwig/hmftools/cup/feature/FeatureDataLoader.java:309]: Another good catch! fixed.

**Review by DFKoetsier — APPROVED** [review]

**Pull request merged**